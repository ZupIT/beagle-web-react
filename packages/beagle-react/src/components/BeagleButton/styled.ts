/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import styled from 'styled-components'
import { BeagleTheme } from '../commons.styled'

export const StyledBeagleButton = styled.button`
  border: 1px solid ${BeagleTheme.swampLight};
  line-height: 34px;
  color: inherit;
  background: transparent;
  box-sizing: border-box;
  position: relative;
  cursor: pointer;
  outline: none;
  display: inline-block;
  white-space: nowrap;
  text-decoration: none;
  vertical-align: baseline;
  text-align: center;
  margin: 0;
  min-width: 70px;
  padding: 0 16px;
  border-radius: 5px;
  overflow: visible;
  
  &:hover {
    background-color: ${BeagleTheme.swampTransparent};
  }

  &:disabled {
    opacity: 0.4;
    pointer-events: none;
  }
`
