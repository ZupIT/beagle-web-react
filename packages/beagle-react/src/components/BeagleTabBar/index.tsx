/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import React, { FC } from 'react'
import { BeagleDefaultComponent } from '../types'
import withTheme from '../utils/withTheme'
import { ImagePath } from '../BeagleImage'
import {
  StyledTabBar,
  StyledBeagleTabItem,
  StyledBeagleTabItemContent,
  StyledBeagleImage,
  StyledSelected,
} from './styled'

export interface ItemTitle {
  title?: string,
  icon?: ImagePath,
}

export interface BeagleTabBarInterface extends BeagleDefaultComponent {
  onTabSelection?: (item: number) => void,
  currentTab?: number,
  items: ItemTitle[],
  styleId?: string,
}

const BeagleTabBar: FC<BeagleTabBarInterface> = ({ onTabSelection, currentTab, items }) => {

  const changeSelectedTab = (index: number) => {
    if (!onTabSelection) return
    onTabSelection(index)
  }

  const tabImage = (item: ItemTitle) => item.icon ? (
    <StyledBeagleImage path={item.icon}></StyledBeagleImage>
  ) : null

  return (
    <StyledTabBar>
      {items.map((item, index) => (
        <StyledBeagleTabItem key={index}>
          <StyledBeagleTabItemContent onClick={() => changeSelectedTab(index)}>
            {tabImage(item)}
            {item.title}
          </StyledBeagleTabItemContent >
          {index === currentTab && <StyledSelected></StyledSelected>}
        </StyledBeagleTabItem>
      ))}
    </StyledTabBar>
  )
}

export default withTheme(BeagleTabBar)
