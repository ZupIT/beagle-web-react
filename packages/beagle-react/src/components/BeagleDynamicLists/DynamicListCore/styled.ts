/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import styled from 'styled-components'
import { Direction } from 'common/models'

interface StyledDynamicViewsInterface {
  direction: Direction,
  useParentScroll?: boolean,
  isScrollIndicatorVisible?: boolean,
  listType?: string,
  spanCount?: number,
}

export const StyledDynamicViewsInterface = styled.div<StyledDynamicViewsInterface>`
  display: ${({ listType }) => listType === 'GRID' ? 'grid' : 'flex'};
  grid-auto-flow: ${({ direction }) => direction === 'HORIZONTAL' ? 'column' : 'row'};
  grid-template-columns: ${({ listType, spanCount, direction }) =>
    (listType === 'GRID' && spanCount && direction === 'VERTICAL') && `repeat(${spanCount}, 1fr)`};
  grid-template-rows: ${({ listType, spanCount, direction }) =>
    (listType === 'GRID' && spanCount && direction === 'HORIZONTAL')
    && `repeat(${spanCount}, 1fr)`};
  justify-content: ${({ listType, direction }) => listType === 'GRID'
    && direction === 'VERTICAL' && 'space-evenly'};
  flex-direction: ${({ direction }) => direction === 'VERTICAL' ? 'column' : 'row'};
  overflow: ${({ useParentScroll }) => useParentScroll ? 'inherit' : 'auto'};
  width: ${({ direction }) => direction === 'HORIZONTAL' ? '100%' : 'auto'};
  height: ${({ direction }) => direction === 'VERTICAL' ? '100%' : 'auto'};
  & ::-webkit-scrollbar {
    display: ${({ isScrollIndicatorVisible }) => isScrollIndicatorVisible ? 'auto' : 'none'};
  }
  -ms-overflow-style: ${({ isScrollIndicatorVisible }) => isScrollIndicatorVisible ?
    'auto' : 'none'};
  scrollbar-width: ${({ isScrollIndicatorVisible }) => isScrollIndicatorVisible ? 'auto' : 'none'};
`
