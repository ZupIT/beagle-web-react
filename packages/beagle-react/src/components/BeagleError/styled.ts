/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import styled from 'styled-components'
import BeagleButton from '../BeagleButton'

export const StyledRetryContainer = styled.section`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  margin: 2em;
  font-size: 23px;
`

export const StyledRetryButton = styled(BeagleButton)`
  max-width: 200px;
  margin: 8px 4px 4px 4px;
`

export const StyledShowMoreButton = styled(BeagleButton)`
  max-width: 200px;
  margin: 4px;
  border: none;
`

export const ErrorsDetailedContainer = styled.section`
  width: 100%;
  max-width: 800px;
  max-height: 0;
  opacity: 0;
  overflow: auto;
  position: relative;
  margin: 4px;
  border-radius: 8px;
  border: 1px solid #FFD7D7;
  background-color: #FFF0F0;
  padding: 8px;
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;
  gap: 8px;
  transition: all 0.2s ease;

  &.show {
    max-height: 1024px;
    opacity: 1;
  }

  .error-message {
    font-family: Verdana, Geneva, Tahoma, sans-serif;
    position: relative;
    color: #FF0000;
    font-size: 14px;
    padding: 0;
    margin: 0;

    .error-indicator {
        color: #414a4c;
        margin: 0 4px 0 0;
        font-size: 12px;
    }
  }
`
