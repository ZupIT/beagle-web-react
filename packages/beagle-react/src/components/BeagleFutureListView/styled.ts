/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import styled from 'styled-components'
import { Direction } from 'common/models'

interface StyledListViewInterface {
  direction: Direction,
  useParentScroll?: boolean,
}

export const StyledListView = styled.div<StyledListViewInterface>`
  display: flex;
  flex-direction: ${({ direction }) => direction === 'VERTICAL' ? 'column' : 'row'};
  overflow: ${({ useParentScroll }) => useParentScroll ? 'inherit' : 'auto'};
  width: ${({ direction }) => direction === 'HORIZONTAL' ? '100%' : 'auto'};
  height: ${({ direction }) => direction === 'VERTICAL' ? '100%' : 'auto'};
`
