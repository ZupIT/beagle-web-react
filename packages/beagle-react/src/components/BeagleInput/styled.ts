/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import styled from 'styled-components'

interface StyledTextInputInterface {
  error?: string,
  showError?: boolean,
}

export const errorColor = '#FF0000'

export const StyledInput = styled.input<StyledTextInputInterface>`
  border: ${({ error, showError }) => error && showError ? errorColor : ''};
  outline: ${({ error, showError }) => error && showError ? `auto ${errorColor}` : ''};
`
