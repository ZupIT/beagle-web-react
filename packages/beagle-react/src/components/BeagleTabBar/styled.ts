/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import styled from 'styled-components'
import { BeagleTheme } from '../commons.styled'
import BeagleImage from '../BeagleImage'

export const StyledTabBar = styled.div`
  display: flex;
  align-items: center;
`

export const StyledBeagleTabItem = styled.div`
  display: flex;
  flex-direction: column;
  margin: 0 10px;
`

export const StyledBeagleTabItemContent = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  min-width: 20px;
  min-height: 15px;

  &:hover {
    background-color: ${BeagleTheme.swampTransparent};
  }
`

export const StyledBeagleImage= styled(BeagleImage) `
  max-width: 80px;
  max-height: 80px;
`

export const StyledSelected = styled.div`
  height: 2px;
  background-color: ${BeagleTheme.swamp};
`
