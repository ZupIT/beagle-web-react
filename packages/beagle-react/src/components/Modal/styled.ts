import styled from 'styled-components'
import { BeagleTheme } from '../commons.styled'

export const StyledModal = styled.div`
  background: ${BeagleTheme.darkGray};
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  z-index: 1;
  display: flex;
  align-items: center;
  justify-content: center;

  > div {
    background-color: #FFF;
    border-radius: 25px;
    padding: 30px;
  }
`
