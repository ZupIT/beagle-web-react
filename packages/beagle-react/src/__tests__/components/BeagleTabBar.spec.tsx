/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import { configure, mount } from 'enzyme'
import BeagleTabBar from '../../components/BeagleTabBar'
import { StyledBeagleTabItem, StyledSelected } from '../../components/BeagleTabBar/styled'

let wrapper: any
let tabIndex: number

configure({ adapter: new Adapter() })

beforeAll(() => {
  const setTabIndex = (e: number) => {
    tabIndex = e
  }

  wrapper = mount(
    <BeagleTabBar
      items={[{ title: 'First Tab' }, {title: 'Second Tab'}]}
      currentTab={tabIndex}
      onTabSelection={setTabIndex}
    />
  )
})

test('Beagle Tab Bar should render the tabs corretly', () => {
  expect(wrapper).toMatchSnapshot()
})

test('Beagle Tab Bar should toggle the tab', () => {
  let tabs = wrapper.find(StyledBeagleTabItem)

  expect(tabs.length).toBe(2)
  expect(tabs.at(0).exists(StyledSelected)).toBe(false)
  expect(tabs.at(1).exists(StyledSelected)).toBe(false)

  tabs.at(1).find('div').at(1).simulate('click')
  wrapper.setProps({ currentTab: tabIndex })
  tabs = wrapper.find(StyledBeagleTabItem)
  expect(tabs.at(0).exists(StyledSelected)).toBe(false)
  expect(tabs.at(1).exists(StyledSelected)).toBe(true)

  tabs.at(0).find('div').at(1).simulate('click')
  wrapper.setProps({ currentTab: tabIndex })
  tabs = wrapper.find(StyledBeagleTabItem)
  expect(tabs.at(0).exists(StyledSelected)).toBe(true)
  expect(tabs.at(1).exists(StyledSelected)).toBe(false)

  tabs.at(1).find('div').at(1).simulate('click')
  wrapper.setProps({ currentTab: tabIndex })
  tabs = wrapper.find(StyledBeagleTabItem)
  expect(tabs.at(0).exists(StyledSelected)).toBe(false)
  expect(tabs.at(1).exists(StyledSelected)).toBe(true)
})


