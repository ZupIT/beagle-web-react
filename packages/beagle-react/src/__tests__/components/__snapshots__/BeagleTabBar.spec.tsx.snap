// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Beagle Tab Bar should render the tabs corretly 1`] = `
<ComponentWithTheme
  items={
    Array [
      Object {
        "title": "First Tab",
      },
      Object {
        "title": "Second Tab",
      },
    ]
  }
  onTabSelection={[Function]}
>
  <BeagleTabBar
    className=""
    items={
      Array [
        Object {
          "title": "First Tab",
        },
        Object {
          "title": "Second Tab",
        },
      ]
    }
    onTabSelection={[Function]}
  >
    <styled.div
      aria-colcount={2}
    >
      <div
        aria-colcount={2}
        className="sc-dlfnbm CZBcd"
      >
        <styled.div
          aria-colindex={0}
          aria-posinset={1}
          aria-selected={false}
          key="0"
        >
          <div
            aria-colindex={0}
            aria-posinset={1}
            aria-selected={false}
            className="sc-hKgILt dVipMb"
          >
            <styled.div
              onClick={[Function]}
            >
              <div
                className="sc-eCssSg beqbLp"
                onClick={[Function]}
              >
                First Tab
              </div>
            </styled.div>
          </div>
        </styled.div>
        <styled.div
          aria-colindex={1}
          aria-posinset={2}
          aria-selected={false}
          key="1"
        >
          <div
            aria-colindex={1}
            aria-posinset={2}
            aria-selected={false}
            className="sc-hKgILt dVipMb"
          >
            <styled.div
              onClick={[Function]}
            >
              <div
                className="sc-eCssSg beqbLp"
                onClick={[Function]}
              >
                Second Tab
              </div>
            </styled.div>
          </div>
        </styled.div>
      </div>
    </styled.div>
  </BeagleTabBar>
</ComponentWithTheme>
`;
